### PR TITLE
feat(ui): add state management on pipeline page

### PR DESCRIPTION
**Pull request by bnjjj**
1. Description
+ add fix about rust sdk
1. Related issues
1. About tests
1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report ut-engine#8602.0 ✘  
* Stage 1
  * Engine Test with DB ✘

File: ui/src/app/views/pipeline/show/workflow/pipeline.workflow.component.ts [hunk 1]
```diff
@@ -171,14 +192,24 @@ export class PipelineWorkflowComponent implements OnInit, OnDestroy {
...
+                this._toast.success('', this._translate.instant('stage_job_added'));
 
-            let currentStage = pip.stages.find((stage) => this.selectedStage.id === stage.id);
-            if (currentStage) {
-                this.selectJob(currentStage.jobs[currentStage.jobs.length - 1], this.selectedStage);
-            }
-        });
+                console.log('selectedStage ID', this.selectedStage.id);
+                let currentStage = pip.stages.find((stage) => this.selectedStage.id === stage.id);
+                console.log(currentStage);
```
Comment (richardlt) [inline ui/src/app/views/pipeline/show/workflow/pipeline.workflow.component.ts]: To remove

File: ui/src/app/views/pipeline/show/workflow/pipeline.workflow.component.ts [hunk 2]
```diff
@@ -171,14 +192,24 @@ export class PipelineWorkflowComponent implements OnInit, OnDestroy {
...
+        )
+            .subscribe((pip) => {
+                this._toast.success('', this._translate.instant('stage_job_added'));
 
-            let currentStage = pip.stages.find((stage) => this.selectedStage.id === stage.id);
-            if (currentStage) {
-                this.selectJob(currentStage.jobs[currentStage.jobs.length - 1], this.selectedStage);
-            }
-        });
+                console.log('selectedStage ID', this.selectedStage.id);
```
Comment (richardlt) [inline ui/src/app/views/pipeline/show/workflow/pipeline.workflow.component.ts]: To remove

File: ui/src/app/views/pipeline/show/workflow/pipeline.workflow.component.ts [hunk 3]
```diff
@@ -160,9 +173,17 @@ export class PipelineWorkflowComponent implements OnInit, OnDestroy {
...
+            projectKey: this.project.key,
+            pipelineName: this.pipeline.name,
+            stage: s
+        })).pipe(
+            flatMap(() => this.store.selectOnce(PipelinesState.selectPipeline(this.project.key, this.pipeline.name)))
+        ).subscribe((pip) => {
             this._toast.success('', this._translate.instant('stage_added'));
-            this.selectedStage = p.stages[p.stages.length - 1];
+            this.selectedStage = pip.stages[pip.stages.length - 1];
+            console.log(this.selectedStage);
```
Comment (richardlt) [inline ui/src/app/views/pipeline/show/workflow/pipeline.workflow.component.ts]: To remove

File: engine/api/pipeline_import.go [hunk 4]
```diff
@@ -6,6 +6,8 @@ import (
 	"io/ioutil"
 	"net/http"
 
+	"github.com/ovh/cds/engine/api/permission"
```
Comment (sguiheux) [inline engine/api/pipeline_import.go]: wrong place

**Pull request merged**